demo/client/backend/src/fabricClient.js

```diff
@@ -0,0 +1,477 @@
+        };
+
+        // Connect to gateway using application specified parameters
+        //  using local variable, userGateway
+        console.log('Connect to Fabric gateway for userName: ' + userName);
+        let userGateway = new Gateway ();
+        await userGateway.connect(connectionProfile, connectionOptions);
+
+        console.log ("Connected to gateway");
+        network =  await userGateway.getNetwork (configdata["channel_name"]);
+        console.log ("Connected to channel, " + configdata["channel_name"]);
+
+        //  contract is global variable, used by invoke and query functions
+        contract =  await network.getContract (configdata["smart_contract_name"]);
+        console.log ("Connected to chaincode, ", configdata["smart_contract_name"]);
+        console.log ('...........................');
+        //  return success
+        var result =
+        {
+          "status": {  'rc': SUCCESS,
+          'msg': 'Successfully set user context to  user:  ' + userName }
+        };
+        return result;
+        }
+    catch (error) {
+        var result =
+        {
+          "status": {  'rc': FAILURE,
+          'msg': 'setUserContext failed for userName,' + userName + ' : ' + error }
+        };
+        console.error('setUserContext failed for userName,' + userName );
+        throw (result);
+      }
+}  //  end of UserContext(userName)
+
+async function invoke(userName, txName, ...args) {
+    console.log (">>> In function, fabricClient:invoke... ");
+    console.log ("userName =***" + userName + "***");
+    console.log ("txName =  ", txName);
+    console.log ("args =  ", args);
+
+    try {
+    await setUserContext (userName,secret);
+    }  catch (error) {
+        throw (error);
+    }
+
+    var result = contract.submitTransaction(txName,...args);
+    return result.then((response) => {
+    // process response
+    console.log('Submitted transaction successfully');
+
+    console.log ('...........................');
+    console.log ("Response(buffer): " ,  response);
+    console.log ('...........................');
+
+    //  convert buffer to string
+    var bufferData = {
+                           "type" : "Buffer",
+                           "data" : []
+                    };
+       bufferData["data"] = response;
+       strResponse = String.fromCharCode.apply (null, bufferData["data"]);
+
+    console.log ("Response (string): ", strResponse);
+    console.log ('...........................');
+
+    //console.log ('If base64 encoded, to decode the response, use ... echo <ResponseData> | base64 -d - ; echo "  "');
+    //console.log ("-------------------------------------");
+
+    return Promise.resolve(strResponse);
+    },(error) =>
+    {
+       //  handle error if transaction failed
+       console.log('Error thrown from tx promise',error);
+       return Promise.reject(error);
+    });
+}
+
+async function query (userName, txName, ...args) {
```

Comment:
this function looks very similar to invoke except of `evaluateTransaction`. Please try to reduce code duplication